File: glucare_java/glucare/src/main/java/com/uade/glucare/model/DiaryEntry.java
package com.uade.glucare.model;

import java.time.LocalDate;
import java.time.LocalTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.util.List;

@Entity
@Table(name = "diary_entry")
public class DiaryEntry {
    @Id
    @GeneratedValue
    @Column(name = "id")
    private Long id;

    @Column(name = "fecha")
    private LocalDate fecha;

    @Column(name = "hora")
    private LocalTime hora;

    @ManyToOne
    @JoinColumn(name = "mood_id")
    private Mood estadoAnimo;

    @Column(name = "medicamentos_tomados")
    @ManyToMany
    private List<Medicine> medicamentosTomados;

    @Column(name = "alimentos_consumidos")
    @ManyToMany
    private List<FoodItem> alimentosConsumidos;

    @ManyToOne
    @JoinColumn(name = "physical_activity_id")
    private PhysicalActivity actividadFisicaRealizada;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    //Getters y setters
    public Long getId() {
        return id;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public LocalTime getHora() {
        return hora;
    }

    public void setHora(LocalTime hora) {
        this.hora = hora;
    }

    public Mood getEstadoAnimo() {
        return estadoAnimo;
    }

    public void setEstadoAnimo(Mood estadoAnimo) {
        this.estadoAnimo = estadoAnimo;
    }

    public List<Medicine> getMedicamentosTomados() {
        return medicamentosTomados;
    }

    public void setMedicamentosTomados(List<Medicine> medicamentosTomados) {
        this.medicamentosTomados = medicamentosTomados;
    }

    public List<FoodItem> getAlimentosConsumidos() {
        return alimentosConsumidos;
    }

    public void setAlimentosConsumidos(List<FoodItem> alimentosConsumidos) {
        this.alimentosConsumidos = alimentosConsumidos;
    }

    public PhysicalActivity getActividadFisicaRealizada() {
        return actividadFisicaRealizada;
    }

    public void setActividadFisicaRealizada(PhysicalActivity actividadFisicaRealizada) {
        this.actividadFisicaRealizada = actividadFisicaRealizada;
    }


   
}

File: glucare_flutter/lib/model/DiaryEntry.dart
import 'package:flutter/material.dart';
import 'package:glucare/model/FoodItem.dart';
import 'package:glucare/model/Medicine.dart';
import 'package:glucare/model/Mood.dart';
import 'package:glucare/model/PhysicalActivity.dart';
import 'package:glucare/model/User.dart';

class DiaryEntry {
  final int id;
  final DateTime fecha;
  final TimeOfDay hora;
  final Mood estadoAnimo;
  final List<Medicine> medicamentosTomados;
  final List<FoodItem> alimentosConsumidos;
  final PhysicalActivity actividadFisicaRealizada;
  final User user;

  DiaryEntry({
    required this.id,
    required this.fecha,
    required this.hora,
    required this.estadoAnimo,
    required this.medicamentosTomados,
    required this.alimentosConsumidos,
    required this.actividadFisicaRealizada,
    required this.user,
  });

  factory DiaryEntry.fromJson(Map<String, dynamic> json) {
    return DiaryEntry(
      id: json['id'],
      fecha: DateTime.parse(json['fecha']),
      hora: TimeOfDay(
        hour: int.parse(json['hora'].split(':')[0]),
        minute: int.parse(json['hora'].split(':')[1]),
      ),
      estadoAnimo: Mood.fromJson(json['estadoAnimo']),
      medicamentosTomados: (json['medicamentosTomados'] as List)
          .map((i) => Medicine.fromJson(i))
          .toList(),
      alimentosConsumidos: (json['alimentosConsumidos'] as List)
          .map((i) => FoodItem.fromJson(i))
          .toList(),
      actividadFisicaRealizada:
          PhysicalActivity.fromJson(json['actividadFisicaRealizada']),
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fecha': fecha.toIso8601String(),
      'hora': '${hora.hour}:${hora.minute}',
      'estadoAnimo': estadoAnimo.toJson(),
      'medicamentosTomados':
          medicamentosTomados.map((medicine) => medicine.toJson()).toList(),
      'alimentosConsumidos':
          alimentosConsumidos.map((foodItem) => foodItem.toJson()).toList(),
      'actividadFisicaRealizada': actividadFisicaRealizada.toJson(),
      'user': user.toJson(),
    };
  }
}
